File: pa3/Sparse.java
/*
    Zachary Dutra
    zdutra
    1581789
    pa3
*/

import java.util.*;
import java.io.*;
class Sparse{
public static void main(String[]args)throws IOException{
    PrintWriter out;
    Scanner in;
    String line;
    int lineNumber = 0;
    int mSize;
    int A_NNZ;
    int B_NNZ;
    int row, column;
    double data;
    if(args.length < 2){
        System.err.println("Usage: FileIO infile outfile");
        System.exit(1);
    }
    FileReader fr = new FileReader(args[0]);
    in = new Scanner(new File(args[0]));
    FileWriter output = new FileWriter(new File(args[1]));
    out = new PrintWriter(output);

    mSize = in.nextInt();

    Matrix A = new Matrix(mSize);
    Matrix B = new Matrix(mSize);

    A_NNZ = in.nextInt();
    B_NNZ = in.nextInt();
    for(int i = 0; i < A_NNZ; i++){
        row = in.nextInt();
        column = in.nextInt();
        data = in.nextDouble();
        A.changeEntry(row, column, data);
    }
    line = in.nextLine();
    for(int i = 0; i < B_NNZ; i++){
        row = in.nextInt();
        column = in.nextInt();
        data = in.nextDouble();
        B.changeEntry(row, column, data);
    }
    out.println("A has " + A_NNZ + " non-zero entries:");
    out.println(A);

    out.println("B has " + B_NNZ + " non-zero entries:");
    out.println(B);

    out.println("(1.5)*A =");
    out.println(A.scalarMult(1.5));

    out.println("A+B =");
    out.println(A.add(B));

    out.println("A+A =");
    out.println(A.add(A));

    out.println("B-A =");
    out.println(B.sub(A));

    out.println("A-A =");
    out.println(A.sub(A));

    out.println("Transpose(A) =");
    out.println(A.transpose());

    out.println("A*B =");
    out.println(A.mult(B));

    out.println("B*B =");
    out.print(B.mult(B));

    in.close();
    output.close();
    out.close();
}
}

File: CSE101/pa2/Sparse.c
#include<stdio.h>
#include<stdlib.h>
#include<string.h>
#include"Matrix.h"

#define MAX_LEN 250

int main(int argc, char* argv[]){
    FILE *in, *out;
    char line[MAX_LEN];
    int mSize;
    int A_NNZ;
    int B_NNZ;
    int row;
    int column;
    double data;
    char* token;
    Matrix A;
    Matrix B;
    // check command line for correct number of arguments
   if( argc != 3 ){
      printf("Usage: %s <input file> <output file>\n", argv[0]);
      exit(1);
   }
 
   // open files for reading and writing 
   in = fopen(argv[1], "r");
   out = fopen(argv[2], "w");
   if( in==NULL ){
      printf("Unable to open file %s for reading\n", argv[1]);
      exit(1);
   }
   if( out==NULL ){
      printf("Unable to open file %s for writing\n", argv[2]);
      exit(1);
   }
   for(int j = 0; fgets(line, MAX_LEN, in) != NULL; j++){
       fprintf(stdout, "%s\n", line);
       token = strtok(line, " \n");
       if(j == 0){
           mSize = atoi(token);
           A = newMatrix(mSize);
           B = newMatrix(mSize);
           token = strtok(NULL, " ");
           A_NNZ = atoi(token);
           token = strtok(NULL, " ");
           B_NNZ = atoi(token);
           continue;
       }
       if(j <= A_NNZ+1 && j >= 2){
        row = atoi(token);
        token = strtok(NULL, " ");
        column = atoi(token);
        token = strtok(NULL, " ");
        data = atoi(token);
        changeEntry(A, row, column, data);
       }
       if(j >= A_NNZ+3 && j <= A_NNZ+B_NNZ+2){
        row = atoi(token);
        token = strtok(NULL, " ");
        column = atoi(token);
        token = strtok(NULL, " ");
        data = atoi(token);
        changeEntry(B, row, column, data);
       }
   }
    fprintf(out, "A has %d non-zero entries:\n", A_NNZ);
    printMatrix(out, A);
    fprintf(out, "B has %d non-zero entries:\n", B_NNZ);
    printMatrix(out, B);
    fprintf(out, "(1.5)*A =\n");
    Matrix C = scalarMult(1.5, A);
    printMatrix(out, C);
    fprintf(out,"A+B =\n");
    C = sum(A,B);
    printMatrix(out, C);
    fprintf(out,"A+A =\n");
    C = sum(A,A);
    printMatrix(out, C);
    fprintf(out,"B-A =\n");
    C = diff(B,A);
    printMatrix(out, C);
    fprintf(out,"A-A =\n");
    C = diff(A,A);
    printMatrix(out, C);
    fprintf(out,"Transpose(A) =\n");
    C = transpose(A);
    printMatrix(out, C);
    fprintf(out,"A*B =\n");
    C = product(A,B);
    printMatrix(out, C);
    fprintf(out,"B*B =\n");
    C = product(B,B);
    printMatrix(out, C);
}